mongodbatlas/data_source_mongodbatlas_search_index.go

```diff
@@ -0,0 +1,73 @@
+package mongodbatlas
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+func dataSourceMongoDBAtlasSearchIndex() *schema.Resource {
+	return &schema.Resource{
+		Read:   dataSourceMongoDBAtlasSearchIndexRead,
+		Schema: returnSearchIndexSchema(),
+	}
+
+}
+
+func dataSourceMongoDBAtlasSearchIndexRead(d *schema.ResourceData, meta interface{}) error {
+	// Get client connection.
+	conn := meta.(*matlas.Client)
```

Comment:
Please get the changes from the master and change this line to `conn := meta.(*MongoDBClient).Atlas` , otherwise once it merges to master it won't work if you don't update the changes from the master